Implements #15, request for hardware concurrency utility function

kdt3rd opened the pull request:
new static member of ThreadPool, call as
ThreadPool::hardwareConcurrency, so no abi breakage or api change

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
Implements #15, request for hardware concurrency utility function

new static member of ThreadPool, call as
ThreadPool::hardwareConcurrency, so no abi breakage or api change

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

lgritz reviewed (commented):
LGTM, but refresh my memory, are we not C++11 minimum moving forward? Why not use std::thread::hardware_concurrency always? Or expect users to do that?

meshula commented:
same question as Larry, see this comment by @cary-ilm - https://github.com/openexr/openexr/pull/443#issuecomment-512005797

axxel commented:
+1 for letting the user use `std::thread::hardware_concurrency` on the client side.

kdt3rd commented:
This function, when in c++11 mode is just the same (and why I called it what I did), clients can use whatever. We haven't removed the old c++98 code yet, and this ticket was on my assigned list, so I did something about it :)

meshula reviewed (approved):
Cool, I'm fine with resolving FORCE_CXX03 across the codebase as a separate task

kdt3rd committed:
Rename new function and clarify purpose

After discussion with phillman, renamed to give this routine a purpose
beyond some soon to be deleted legacy support, and clarified this in the
comment documenting the function.

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>
